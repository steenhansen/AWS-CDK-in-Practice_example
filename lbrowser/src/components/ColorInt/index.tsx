import React from 'react';

import { Interfaces } from '../../../program.interfaces';

import { UserBox, UserContainer, UserContent } from './styles';

interface Props {
  color_int: Interfaces.ColorInt;
}


export const ColorInt: React.FC<Props> = ({ color_int }) => {
  return (
    <UserContainer key={color_int.the_color}>
      <UserBox>
        <UserContent>
          <li >
            {color_int.the_color} - {color_int.the_integer}
          </li>
        </UserContent>
      </UserBox>
    </UserContainer>
  );
};

