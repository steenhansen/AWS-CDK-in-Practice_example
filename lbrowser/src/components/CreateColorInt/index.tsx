import React, { useState } from 'react';
import { Interfaces } from '../../../program.interfaces';

import { CreateColorIntContainer } from './styles';

interface Props {
  handleAdd: ({
    new_color_int,
  }: {
    new_color_int: Interfaces.ColorInt;
  }) => Promise<void>;

  handleClear: () => Promise<void>;
}

export const CreateColorInt: React.FC<Props> = ({ handleAdd, handleClear }) => {

  const [new_color_int, setNewColorInt] = useState<Interfaces.ColorInt>({
    id: '',
    the_color: 'purple',
    the_integer: 49
  });

  const handleColorIntChange = (type: string, value: string) => {
    setNewColorInt(current_color_int => ({ ...current_color_int, [type]: value }));
  };






  return (
    <CreateColorIntContainer>
      <span>  red/green/blue</span>&nbsp;
      <input
        autoComplete="off"
        pattern="red|green|blue"
        onFocus={({ target }) => target.value = ""}
        onChange={({ target }) => handleColorIntChange('the_color', target.value)}
        type="text"
        name="new_color_int"
        id="new_color_int"
        placeholder="red, green, or blue"
      />

      <br /><br />
      <span> 1/2/3</span>&nbsp;

      <input
        autoComplete="off"
        pattern="1|2|3"
        onFocus={({ target }) => target.value = ""}
        onChange={({ target }) => handleColorIntChange('the_integer', target.value)}
        type="text"
        name="new_color_int"
        id="new_color_int"
        placeholder="1, 2, or 3"
      />

      <br />  <br />

      <button type="button" onClick={() => handleAdd({ new_color_int })}>
        Add Object to Dynamo
      </button>

      &nbsp;&nbsp;&nbsp;
      <button type="button" onClick={() => handleClear()}>
        Clear Dynamo Table
      </button>

    </CreateColorIntContainer>
  );


};
