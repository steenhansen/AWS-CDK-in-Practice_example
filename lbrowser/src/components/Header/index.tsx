import React from 'react';

import { HeaderContainer } from './styles';


let text_logo: string;
if (process.env["REACT_APP__LOCAL_MODE"] === 'yes') {
  text_logo = 'On Local';
} else {
  text_logo = 'On AWS';
}

export const Header: React.FC = () => {
  return (
    <HeaderContainer>
      <div className="brand">
        <a href="/" data-testid={"header-logo"}>{text_logo}</a> a1
      </div>
    </HeaderContainer>
  );
};
