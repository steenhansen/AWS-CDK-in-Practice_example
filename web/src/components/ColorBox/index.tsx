import React from 'react';

import { ColorInt } from '../../../shapes';


import { UserBox, UserContainer, UserContent } from './styles';

interface BoxProps {
  color_int: ColorInt;
}


export const ColorBox: React.FC<BoxProps> = ({ color_int }) => {
  return (
    <UserContainer key={color_int.the_color}>
      <UserBox>
        <UserContent>
          <li data-testid={"test-" + color_int.the_color}>
            {color_int.the_color} - {color_int.the_integer}
          </li>
        </UserContent>
      </UserBox>
    </UserContainer>
  );
};

