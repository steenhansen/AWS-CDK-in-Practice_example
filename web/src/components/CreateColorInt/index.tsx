import React, { useState } from 'react';
import { Interfaces } from '../../../program.interfaces';

import { CreateColorIntContainer } from './styles';
//import the_constants from '../../../../cicd/program.constants.json';


interface Props {
  handleAdd: ({
    new_color_int,
  }: {
    new_color_int: Interfaces.ColorInt;
  }) => Promise<void>;

  handleClear: () => Promise<void>;
}

export const CreateColorInt: React.FC<Props> = ({ handleAdd, handleClear }) => {

  const [new_color_int, setNewColorInt] = useState<Interfaces.ColorInt>({
    id: '',
    the_color: 'purple',
    the_integer: 49
  });

  const handleColorIntChange = (type: string, value: string) => {
    setNewColorInt(current_color_int => ({ ...current_color_int, [type]: value }));
  };




  return (
    <CreateColorIntContainer>
      <span>  red/green/blue</span>&nbsp;
      <input
        autoComplete="off"

        onFocus={({ target }) => target.value = ""}
        onChange={({ target }) => handleColorIntChange('the_color', target.value)}
        type="text"
        name="new_color_int"
        id="new_color_int"
        data-testid={"test-color_rgb"}
        placeholder="red, green, or blue"
      />

      <br /><br />
      <span> 0...255</span>&nbsp;

      <input
        autoComplete="off"
        onFocus={({ target }) => target.value = ""}
        onChange={({ target }) => handleColorIntChange('the_integer', target.value)}
        type="text"
        name="new_color_int"
        id="new_color_int"
        data-testid={"test-color_integer"}
        placeholder="0...255"
      />

      <br />  <br />

      <button type="button" data-testid="test-change" onClick={() => handleAdd({ new_color_int })}>AddChange</button>

      &nbsp;&nbsp;&nbsp;
      <button type="button"
        onClick={() => handleClear()}
        data-testid="test-clear">
        Clear all 3 rgb color indexes
      </button>

    </CreateColorIntContainer>
  );


};
