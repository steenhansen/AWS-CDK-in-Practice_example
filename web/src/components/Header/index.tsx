import React from 'react';

import { HeaderContainer } from './styles';

export const Header: React.FC = () => {
  return (
    <HeaderContainer>
      <div className="brand">
        <a href="/" data-testid={"header-logo"}>abcdef 22 879</a>
      </div>
    </HeaderContainer>
  );
};
