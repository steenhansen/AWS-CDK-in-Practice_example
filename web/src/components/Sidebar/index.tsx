import React from 'react';
import { SidebarContainer } from './styles';

const { C_cicd_web_AWS_Env_prd_dvl
} = require('../../../program.web_values.json');





export const Sidebar: React.FunctionComponent = () => {
  return <SidebarContainer>
    [Env_prd]
    <br />
    <br />
    Environment
    <br />
    {C_cicd_web_AWS_Env_prd_dvl}

  </SidebarContainer>;
};