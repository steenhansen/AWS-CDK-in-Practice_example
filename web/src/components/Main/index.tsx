import { getDbRgb, putDbRgb, clearDbRgb, currentRGB, getApiUrl, colorValues } from './clear';
import React, { useEffect, useState } from 'react';
import { Interfaces } from '../../../program.interfaces';
import { CreateColorInt } from '../CreateColorInt';
import { MainContainer, BoxedColor, Zxc } from './styles';
import program_config from '../../../../cicd/program.config.json';
import the_constants from '../../../../cicd/program.constants.json';

const PORT_SERVER = the_constants.PORT_SERVER;
const CLEARDB_SLUG = the_constants.CLEARDB_SLUG;
let DOMAIN_NAME = program_config.DOMAIN_NAME;

const [SSM_SLACK_WEBHOOK, backend_url] = getApiUrl();

const handle_clear = `${backend_url}/${CLEARDB_SLUG}`;

export const Main: React.FC = () => {

  // const initial_state = await getDbRgb(backend_url);

  const [color_ints, setUserDatas] = useState<Interfaces.ColorInt[]>([]);
  useEffect(() => {
    const fetchColorInts = async () => {
      let all_data = await getDbRgb(backend_url);
      setUserDatas(all_data);
    };
    fetchColorInts();
  }, []);

  const handleAdd = async ({ new_color_int, }: { new_color_int: Interfaces.ColorInt; }) => {
    let new_obj = await putDbRgb({ new_color_int, backend_url });
    const trunc_list = color_ints.filter(an_object => an_object.the_color !== new_obj.the_color);
    const new_list = [...trunc_list, new_obj];
    setUserDatas(_ => new_list);
  };

  const handleClear = async () => {
    await clearDbRgb(handle_clear);
    setUserDatas([
      { id: 'clear-red', the_color: 'red', the_integer: 0 },
      { id: 'clear-green', the_color: 'green', the_integer: 0 },
      { id: 'clear-blue', the_color: 'blue', the_integer: 0 }
    ]);
  };

  const rgb_statment = currentRGB(color_ints);
  const rgb_styles = { backgroundColor: rgb_statment };
  const list_of_objects: Array<React.JSX.Element> = colorValues(color_ints);

  return (
    <MainContainer >
      <h1>DynamoDB Values </h1>

      ssm <h2>{SSM_SLACK_WEBHOOK}</h2>
      constant<h2>{PORT_SERVER}</h2>

      config<h2>{DOMAIN_NAME}</h2>

      <BoxedColor style={rgb_styles} id={"the-color_box"}>
        <Zxc data-testid={"test-color_box"}  >
          {rgb_statment}
        </Zxc>
      </BoxedColor>
      <br></br>
      <CreateColorInt handleAdd={handleAdd} handleClear={handleClear} />


      <ul onChange={e => console.log("e", e)} data-testid={"test-c_id"} id={"col_box_id"}> {list_of_objects}</ul>
    </MainContainer >
  );





};
