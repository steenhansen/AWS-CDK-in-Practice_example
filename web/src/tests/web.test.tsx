import React from 'react';         // NB if not present  -    ReferenceError: React is not defined
import { App } from '../components/App';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import the_config from '../../../cdk/program.config.json';
const TESTING_ALIVE = the_config.TESTING_ALIVE;

import { render, screen, waitFor } from '@testing-library/react';
if (TESTING_ALIVE !== 'yes') {
  it('health check route 2', () => {
    expect("at-least-one-web-test").toBe("at-least-one-web-test");
  });
} else {

  if (process.env["REACT_APP__LOCAL_MODE"] === 'yes') {
    it('Matches the snapshot.', () => {
      const ui_result = render(<App />);
      expect(ui_result).toMatchSnapshot();
    });

    it('clear colors 0 0 0', async () => {
      render(<App />);
      await userEvent.click(screen.getByTestId('test-clear'));
      await screen.findByTestId('test-color_box');
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(0 0 0)');
      });

      await userEvent.type(screen.getByTestId('test-color_rgb'), 'red');
      await userEvent.type(screen.getByTestId('test-color_integer'), '11');
      await userEvent.click(screen.getByTestId('test-change'));
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(11 0 0)');
      });

      await userEvent.type(screen.getByTestId('test-color_rgb'), 'green');
      await userEvent.type(screen.getByTestId('test-color_integer'), '12');
      await userEvent.click(screen.getByTestId('test-change'));
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(11 12 0)');
      });

      await userEvent.type(screen.getByTestId('test-color_rgb'), 'blue');
      await userEvent.type(screen.getByTestId('test-color_integer'), '13');
      await userEvent.click(screen.getByTestId('test-change'));
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(11 12 13)');
      });

      await userEvent.type(screen.getByTestId('test-color_rgb'), 'red');
      await userEvent.type(screen.getByTestId('test-color_integer'), '88');
      await userEvent.click(screen.getByTestId('test-change'));
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(88 12 13)');
      });

      await userEvent.click(screen.getByTestId('test-clear'));
      await screen.findByTestId('test-color_box');
      await waitFor(() => {
        expect(screen.getByTestId('test-color_box')).toHaveTextContent('rgb(0 0 0)');
      });

      expect(screen.getByTestId('test-color_box')).not.toHaveTextContent('ensure-tests-just-not-blindly-passing-as-sync_funcs');

    });

  } else {         // test in Front-End-Test
    test('renders the  logo', () => {
      render(<App />);

      const westpointLogo = screen.getByTestId('header-logo');

      expect(westpointLogo).toBeInTheDocument();
    });
  }
}









