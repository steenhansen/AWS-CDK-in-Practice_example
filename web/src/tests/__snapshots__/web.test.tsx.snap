// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-knesRu bUXVrb"
      >
        <header
          class="sc-braxZu hkGCUc"
        >
          <div
            class="brand"
          >
            <a
              data-testid="header-logo"
              href="/"
            >
              abcdef 22
            </a>
          </div>
        </header>
        <section
          class="sc-dnaUSb gmCehD"
        >
          <nav
            class="sc-gJhJTp iPONKj"
          >
             Color
          </nav>
          <header
            class="sc-gnpbhQ bhddCY"
          >
            <h1>
              DynamoDB Values 
            </h1>
            ssm 
            <h2>
              un-defined
            </h2>
            constant
            <h2>
              3003
            </h2>
            config
            <h2>
              steenhansen.click
            </h2>
            <nav
              class="sc-la-DxNn eRNNIO"
              id="the-color_box"
            >
              <nav
                class="sc-iCZwEW liqqKe"
                data-testid="test-color_box"
              >
                rgb(0 0 0)
              </nav>
            </nav>
            <div
              class="sc-elDIKY gLwXhs"
            >
              <span>
                  red/green/blue
              </span>
               
              <input
                autocomplete="off"
                data-testid="test-color_rgb"
                id="new_color_int"
                name="new_color_int"
                placeholder="red, green, or blue"
                type="text"
              />
              <br />
              <br />
              <span>
                 0...255
              </span>
               
              <input
                autocomplete="off"
                data-testid="test-color_integer"
                id="new_color_int"
                name="new_color_int"
                placeholder="0...255"
                type="text"
              />
              <br />
                
              <br />
              <button
                data-testid="test-change"
                type="button"
              >
                AddChange
              </button>
                 
              <button
                data-testid="test-clear"
                type="button"
              >
                Clear all 3 rgb color indexes
              </button>
            </div>
            <ul
              data-testid="test-c_id"
              id="col_box_id"
            >
               
            </ul>
          </header>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-knesRu bUXVrb"
    >
      <header
        class="sc-braxZu hkGCUc"
      >
        <div
          class="brand"
        >
          <a
            data-testid="header-logo"
            href="/"
          >
            abcdef 22
          </a>
        </div>
      </header>
      <section
        class="sc-dnaUSb gmCehD"
      >
        <nav
          class="sc-gJhJTp iPONKj"
        >
           Color
        </nav>
        <header
          class="sc-gnpbhQ bhddCY"
        >
          <h1>
            DynamoDB Values 
          </h1>
          ssm 
          <h2>
            un-defined
          </h2>
          constant
          <h2>
            3003
          </h2>
          config
          <h2>
            steenhansen.click
          </h2>
          <nav
            class="sc-la-DxNn eRNNIO"
            id="the-color_box"
          >
            <nav
              class="sc-iCZwEW liqqKe"
              data-testid="test-color_box"
            >
              rgb(0 0 0)
            </nav>
          </nav>
          <div
            class="sc-elDIKY gLwXhs"
          >
            <span>
                red/green/blue
            </span>
             
            <input
              autocomplete="off"
              data-testid="test-color_rgb"
              id="new_color_int"
              name="new_color_int"
              placeholder="red, green, or blue"
              type="text"
            />
            <br />
            <br />
            <span>
               0...255
            </span>
             
            <input
              autocomplete="off"
              data-testid="test-color_integer"
              id="new_color_int"
              name="new_color_int"
              placeholder="0...255"
              type="text"
            />
            <br />
              
            <br />
            <button
              data-testid="test-change"
              type="button"
            >
              AddChange
            </button>
               
            <button
              data-testid="test-clear"
              type="button"
            >
              Clear all 3 rgb color indexes
            </button>
          </div>
          <ul
            data-testid="test-c_id"
            id="col_box_id"
          >
             
          </ul>
        </header>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
